docs/DeveloperGuide.md

```diff
@@ -1,38 +1,1061 @@
+
+**Step 1.** The Parser extracts the help command from the user input and calls the parseHelpCommand method.
+
+**Step 2.** The execute method of the returned HelpCommand object is then called.
+
+**Step 3.** In the execute method, the printHelp method of the UI Class is called and a fixed string displaying all 
+            available commands is called.
+
+
+#### Exit/Bye Command
+
+##### Implementation
+
+**Step 1.** The Parser extracts the bye command from the user input and calls the parseHelpCommand method.
+
+**Step 2.** An ExitCommand is returned and since this command is the only command object with the isExit attribute set
+            as "true", the while loop of the program will terminate since !isExit is the loop condition.
+
+<div style="text-align: right;">
+   <a href="#table-of-contents"> Back to Table of Contents </a>
+</div>
+<!-- @@author -->
+
+<!-- ### UI
+
+<div style="text-align: right;">
+   <a href="#table-of-contents"> Back to Table of Contents </a>
+</div> -->
+
+<!-- @@author jinxuan-owyong -->
+## Backend
+
+The backend uses a simplified RESTful API approach. This allows us to decouple code using the proven industry practices. The following diagram illustrates the relationship between various classes involved in `Backend` as described in the [application architecture](#architecture)
+
+![Backend](./static/backend/BackendClassDiagram.png)
+
+
+To find out more, visit the following sections:
+- [Storage](#storage)
+- [API](#api)
+  - [Add, modify, view or delete an entry - `GET`](#add-modify-view-or-delete-an-entry)
+  - [Access all entries available - `DELETE`, `GET`, `PATCH`, `POST`](#access-all-entries-available)
+
+<div style="text-align: right;">
+   <a href="#table-of-contents"> Back to Table of Contents </a>
+</div>
+
+<!-- @@author nghochi123 -->
+### Storage
+
+The `Storage` class is responsible for the serialization of `Entry` data into a csv-like syntax, as well as the deserialization of that data back into `Entry` objects.
+
+The main callable functions to be used are:
+
+- `readFromDatabase()` - Deserializes data stored in text form back into `Entry` objects. Executed when PocketPal is instantiated
+- `writeToDatabase()` - Serializes `Entry` objects in `EntryLog` into text form.
+- `reset()` - Clears whatever is in the stored text file, without affecting what is in the current `EntryLog`.
+
+The structure of the Storage class is as follows:
+
+![StorageClassDiagram](static/backend/storage/StorageClassDiagram.png)
+
+The Sequence Diagram below illustrates the interactions within the `Parser` component upon initialization of PocketPal, as well as whenever data is being saved.
+
+![StorageSequenceDiagram](static/backend/storage/StorageSequenceDiagram.png) 
+
+<div style="text-align: right;">
+   <a href="#table-of-contents"> Back to Table of Contents </a>
+</div>
+<!-- @@author -->
+
+<!-- @@author jinxuan-owyong -->
+### API
+#### Endpoints
+
+![Endpoints](static/backend/endpoint/EndpointClassDiagram.png)
```

Comment:
Try to minimise the number of intersecting/criss-crossing arrows in the diagram to make it neater and more readable.